source/config/__init__.py

```diff
@@ -727,13 +727,15 @@ def reset(self, factoryDefaults=False):
 		self._initBaseConf(factoryDefaults=factoryDefaults)
 		post_configReset.notify(factoryDefaults=factoryDefaults)
 
-	def createProfile(self, name):
+	def createProfile(self, name: str) -> None:
 		"""Create a profile.
 		@param name: The name of the profile to create.
-		@type name: str
 		@raise ValueError: If a profile with this name already exists.
 		"""
 		if globalVars.appArgs.secure:

```

Comment:
The same code pattern is duplicated many times. Why not create a helper function:

```
def checkSecureMode():
    if globalVars.appArgs.secure:
        ui.message('Unavailable in secure context.')
        return True
    return False

```

and then use:

```
def functionThatShouldNotRunInSecureMode():
    if checkSecureMode():
        return
    # Normal function actions
```

Reply:
A `secure` decorator (or perhaps `not-secure`, we can come up with clear wording later...) is a good idea.

Reply:
I've created a block action decorator in 4cb8736. `@secure` could alias the decorator for secure contexts.
I still need to document the decorator and module further.